server/internal/controllers/users.go

```diff
@@ -261,3 +261,40 @@ func UserDelete(c *gin.Context) {
 
 	db.Delete(&models.User{}, userID)
 }
+
+func Purge(c *gin.Context) {
+	db := getDB(c)
+
+	var query struct {
+		UserUID string `form:"user_uid" binding:"required,uuid"`
+	}
+	if err := c.ShouldBindQuery(&query); err != nil {
+		gin_utils.GinAbortWithErrorBody(c, http.StatusBadRequest, fmt.Errorf("err: %v, uri: %v", err, query))
+		return
+	}
+
+	var userID uint
+	db.Raw("SELECT id FROM users WHERE uid = ? LIMIT 1", query.UserUID).Scan(&userID)
+	if userID == 0 {
+		gin_utils.GinAbortWithErrorBody(c, http.StatusBadRequest, errors.New("User is not found"))
+		return
+	}
+
+	var email string
+	db.Raw("SELECT email FROM users WHERE uid = ? LIMIT 1", query.UserUID).Scan(&email)
+
+	var testChain uint
+	db.Raw("SELECT chain_id FROM user_chains Where user_id = ? AND is_chain_admin = 1", userID).Scan(&testChain)
+	var adminCount uint
+	db.Raw("SELECT COUNT(*) FROM `user_chains` WHERE chain_id = ? AND is_chain_admin = 1", testChain).Scan(&adminCount)
+	if adminCount <= 1 {
+		db.Where("chain_id = ?", testChain).Delete(&models.UserChain{})
+		db.Where("id = ?", testChain).Delete(&models.Chain{})
+	} else {
+		db.Where("user_id = ?", userID).Delete(&models.UserChain{})
+	}
+	db.Where("user_id = ?", userID).Delete(&models.UserToken{})
```

Comment:
@jnorlando  I'm trying where possible to keep away from using the orm and showing raw sql queries instead.
Please use the `db.Exec("")`.

This also evades the soft deletion available in some tables (see `deleted_at`)

Reply:
Why did you not use this sql query instead? To my knowladge it is generally more efficient to use a single db query than multiple.

```sql
DELETE FROM chains
WHERE id IN (
	SELECT uc.chain_id 
	FROM  user_chains AS uc
	GROUP BY uc.chain_id 
	HAVING COUNT(uc.id) = 1 AND uc.chain_id IN (
		SELECT uc2.chain_id
		FROM user_chains AS uc2
		WHERE uc2.is_chain_admin = TRUE AND uc2.user_id = ?
	)
)
```